include/deal.II/base/table_indices.h

```diff
@@ -140,22 +140,30 @@ TableIndices<N>::TableIndices(const unsigned int index0,
 
   switch (N)
     {
-    case 1: // fallthrough
+    case 1:
       Assert (index1 == numbers::invalid_unsigned_int, ExcMessage("more than N index values provided"));
-    case 2: // fallthrough
+      DEAL_II_FALLTHROUGH
```

Comment:
To make the life of IDEs easier, what would you think of putting the semicolon here instead of the definition of the macro? If the macro is defined as empty, then this just yields an empty statement. If it is defined as something nonempty, it results in the same as right now. In either case, anything that doesn't know that it's a macro will think of it as a regular statement and indent it correctly.